File: 0050.pow/solution.py
class Solution:
    def my_pow(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        if n < 0:
            return 1 / self.my_pow(x, -n)
        if n % 2:
            return x * self.my_pow(x, n - 1)
        return self.my_pow(x * x, n // 2)

File: 0050.pow/solution.go
package main

func myPow(x float64, n int) float64 {
	if n == 0 {
		return 1
	}
	if n < 0 {
		return 1 / myPow(x, -n)
	}
	if n%2 == 1 {
		return myPow(x, n-1) * x
	}
	return myPow(x*x, n/2)
}
